src/main/java/games/diamant/DiamantGameState.java

```diff
@@ -0,0 +1,253 @@
+package games.diamant;
+
+import core.AbstractGameState;
+import core.AbstractParameters;
+import core.components.Component;
+import core.components.Deck;
+import core.interfaces.IPrintable;
+import core.turnorders.AlternatingTurnOrder;
+import games.diamant.actions.ContinueInCave;
+import games.diamant.actions.ExitFromCave;
+import games.diamant.actions.OutOfCave;
+import games.diamant.cards.DiamantCard;
+import games.diamant.components.ActionsPlayed;
+import games.diamant.components.DiamantTreasureChest;
+import games.diamant.components.DiamantHand;
+
+import java.util.*;
+
+import static core.CoreConstants.PARTIAL_OBSERVABLE;
+
+public class DiamantGameState extends AbstractGameState implements IPrintable {
+    Deck<DiamantCard>          mainDeck;
+    Deck<DiamantCard>          discardDeck;
+    Deck<DiamantCard>          path;
+
+    List<DiamantTreasureChest> treasureChests;
```

Comment:
Following on from previous comment, is there any difference (apart from name) between DiamantHands and TreasureChests?